<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Impulse taper maker v0.3 (21.07.2020)</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="ConnectionBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>201</width>
      <height>121</height>
     </rect>
    </property>
    <property name="title">
     <string>Connection</string>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>19</y>
       <width>181</width>
       <height>131</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="ConnectionLayout">
      <item>
       <widget class="QPushButton" name="AutoDetectButton">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>AutoDetect</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="ManualConnectionLayout">
        <property name="verticalSpacing">
         <number>6</number>
        </property>
        <item row="0" column="0">
         <widget class="QCheckBox" name="manualConnectionBox">
          <property name="text">
           <string>Manual</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="laserPortLabel">
          <property name="text">
           <string>Laser port:    COM</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="laserPortLineEdit">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="clearButtonEnabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="shutterPortLabel">
          <property name="text">
           <string>Shutter port: COM</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="shutterPortLineEdit">
          <property name="enabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QPushButton" name="manualConnectButton">
          <property name="text">
           <string>Connect</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="LogBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>440</y>
      <width>761</width>
      <height>131</height>
     </rect>
    </property>
    <property name="title">
     <string>Log</string>
    </property>
    <widget class="QTextBrowser" name="LogField">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>741</width>
       <height>101</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="ParametersBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>300</y>
      <width>201</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>Parameters</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>181</width>
       <height>131</height>
      </rect>
     </property>
     <layout class="QFormLayout" name="ParametersLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="fileLabel">
        <property name="text">
         <string>Coordinates File:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="powerLabel">
        <property name="text">
         <string>Laser Power:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="powerLayout">
        <item>
         <widget class="QDoubleSpinBox" name="powerSpinBox">
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="singleStep">
           <double>0.500000000000000</double>
          </property>
          <property name="value">
           <double>10.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="powerPercentLabel">
          <property name="text">
           <string>%</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="openTimeLabel">
        <property name="text">
         <string>Shutter open time:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="openTimeLayout">
        <item>
         <widget class="QSpinBox" name="openSpinBox">
          <property name="maximum">
           <number>999999</number>
          </property>
          <property name="singleStep">
           <number>100</number>
          </property>
          <property name="value">
           <number>1000</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="openTimeMsLabel">
          <property name="text">
           <string>ms</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="1">
       <layout class="QHBoxLayout" name="periodLayout">
        <item>
         <widget class="QSpinBox" name="periodSpinBox">
          <property name="maximum">
           <number>999999</number>
          </property>
          <property name="singleStep">
           <number>100</number>
          </property>
          <property name="value">
           <number>2000</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="periodMsLabel">
          <property name="text">
           <string>ms</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="periodLabel">
        <property name="text">
         <string>Shutter period:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QPushButton" name="startButton">
        <property name="text">
         <string>Start/Stop</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QPushButton" name="saveButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Save config</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="coordinatesLayout">
        <item>
         <layout class="QHBoxLayout" name="fileButtonLayout">
          <item>
           <widget class="QLineEdit" name="fileEdit">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Select file</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QToolButton" name="fileButton">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>30</y>
      <width>531</width>
      <height>411</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Tab 1</string>
     </attribute>
     <widget class="QTableWidget" name="tableWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>251</width>
        <height>371</height>
       </rect>
      </property>
     </widget>
     <widget class="QWidget" name="horizontalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>10</y>
        <width>211</width>
        <height>31</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="rowNumberLabel">
         <property name="text">
          <string>Number of rows:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="rowNumberBox">
         <property name="value">
          <number>12</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="rowsApplyButton">
         <property name="text">
          <string>Apply</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Tab 2</string>
     </attribute>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>-9</y>
        <width>541</width>
        <height>401</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="StagesConrtolBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>180</y>
      <width>201</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>Stages conrtrolling</string>
    </property>
    <widget class="QWidget" name="gridLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>181</width>
       <height>83</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="StagesLayout">
      <item row="0" column="0" colspan="2">
       <widget class="QPushButton" name="StagesToZerosButton">
        <property name="text">
         <string>Stages to zeros</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="MoveStagesField"/>
      </item>
      <item row="2" column="0">
       <widget class="QPushButton" name="MoveStagesButton">
        <property name="text">
         <string>Move stages</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QPushButton" name="StagesToHomeButton">
        <property name="text">
         <string>Stages to home</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>AutoDetectButton</tabstop>
  <tabstop>manualConnectionBox</tabstop>
  <tabstop>shutterPortLineEdit</tabstop>
  <tabstop>manualConnectButton</tabstop>
  <tabstop>LogField</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
